crypto/nrf_cc312_platform/include/nrf_cc3xx_platform_shadow_key.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+/**@file
+ * @defgroup nrf_cc3xx_platform_shadow_key nrf_cc3xx_platform derived key APIs
+ * @ingroup nrf_cc3xx_platform
+ * @{
+ * @brief The nrf_cc3xx_platform_shadow_key APIs provide functions helping with
+ *        encrypting/decrypting data using derived keys from the KMU/KDR
+ */
+#include <stdint.h>
+
+#ifndef NRF_CC3XX_PLATFORM_SHADOW_KEY__
+#define NRF_CC3XX_PLATFORM_SHADOW_KEY__
+
+#define NRF_CC3XX_SHADOW_KEY_MAX_INPLACE_CHUNK_SIZE (512)
+#define NRF_CC3XX_SHADOW_AEAD_CONTEXT_SIZE_WORDS (66)
+
+typedef enum {
+    MODE_ENCRYPT,
+    MODE_DECRYPT
+} nrf_cc3xx_platform_shadow_key_enc_mode_t;
+
+/** @brief Enum with the encryption/decryption algorithms supported with derived keys */
+typedef enum {
+    ALG_AES_128_BIT         = 1,
+    ALG_AES_256_BIT         = 2,
+    ALG_CHACHA20_256_BIT    = 3,
+    ALG_CHACHAPOLY_256_BIT  = 3
+} nrf_cc3xx_platform_shadow_key_enc_cipher_t;
+
+/** @brief Structure containing information for doing the key derivation using AES CMAC */
+typedef struct{
+    uint32_t            slot_id;                                     /**< The slot id containing the AES key used in the derivation process. */
+    size_t              key_size;                                    /**< The size of the AES key in bits. */
+    uint8_t const       *label;                                      /**< Buffer containing the label used in the derivation process. */
```

Comment:
```suggestion
    uint8_t const       *label;                                      /**< Pointer to the label used in the derivation process. */
```

Buffer implies mutability.